app/services/designation-service.js

```diff
@@ -171,6 +176,108 @@ export default ApiBaseService.extend(NavigationAwareness, {
     return this.PUT(url, params);
   },
 
+  async resolveOrder(order) {
+    if (order) {
+      return order;
+    }
+    return this.userPickOrder({
+      state: ACTIVE_ORDER_STATES.join(",")
+    });
+  },
+
+  beginDesignation(options) {
+    let pkg = options.pkg,
+      order = options.order;
+
+    this.set("allowItemChange", options.allowItemChange);
+    this.set("allowOrderChange", options.allowOrderChange);
+
+    if (pkg.get("availableQuantity") > 0 || options.isEditOperation) {
+      this.triggerDesignateAction(pkg, order);
+    } else {
+      this.set("designationTargetPackage", pkg);
+      this.set("displayDesignateForm", true);
+      this.set("showItemAvailability", true);
+    }
+  },
+
+  triggerDesignateAction(pkg, order) {
+    this.resolveOrder(order).then(target => {
```

Comment:
Assuming that `resolveOrder` just brings up an overlay, I think it will be better to do 
```
const target = await this.resolveOrder(order)
if(!target) return
// rest of the logic

```